### PR TITLE
[pigeon] Minor C++ output adjustments

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Fixes a stray ; on an inline destructor.
- Changes serialization to `push_back` instead of initializer lists since it turns out the latter prevents move optimizations.
- Removes `flutter::` from a number of common types in the implementation file. (This is something that would have been complicated before, but in the new two-class structure it's very easy to see which code needs `flutter::` and which doesn't by what class it's in.)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pigeon/test/cpp_generator_test.dart [hunk 1]
```diff
@@ -57,6 +57,7 @@ void main() {
       expect(code, contains('class Input'));
       expect(code, contains('class Output'));
       expect(code, contains('class Api'));
+      expect(code, contains('virtual ~Api() {}'));
```
Comment (tarrinneal) [inline packages/pigeon/test/cpp_generator_test.dart:60]: If this is the stray semi-colon being removed, this test doesn't cover that properly.

File: packages/pigeon/lib/cpp_generator.dart [hunk 2]
```diff
@@ -978,6 +991,96 @@ ${prefix}reply(flutter::EncodableValue(std::move(wrapped)));''';
...
+      encodableValue = 'EncodableValue($operator$variableName)';
+    }
+
+    if (hostType.isNullable) {
+      return '$variableName ? $encodableValue : EncodableValue()';
+    }
+    return encodableValue;
+  }
+
+  // Writes the code to declare and populate a variable of type [hostType]
```
Comment (tarrinneal) [inline packages/pigeon/lib/cpp_generator.dart:1021]: should be `///`

File: packages/pigeon/test/cpp_generator_test.dart (hunk 1 above)
Comment (stuartmorgan) [reply packages/pigeon/test/cpp_generator_test.dart:60]: Oops, good point

**Review by tarrinneal — APPROVED** [review]
code looks a lot better without all the flutter::'s, two nits and we're good to go. Once the tests are passing anyway.

**Comment by stuartmorgan** [comment]
The test failures were just me forgetting to update the version in the generator 🎉

**Pull request merged**